openassessment/data.py

```diff
@@ -95,6 +95,48 @@ def map_anonymized_ids_to_usernames(anonymized_ids):
 
     return anonymous_id_to_username_mapping
 
+def map_anonymized_ids_to_emails(anonymized_ids):
```

Comment:
Here's an idea, rather than having these all be separate, and adding a new function every time we want more user data, why not do something like
```
def map_anonymized_ids_to_user_data(anonymized_ids):
    """
    Args:
        anonymized_ids - list of anonymized user ids.
    Returns:
        dict {
           <anonymous_user_id> : {
               'email': (str) <user.email>
               'username': (str) <user.username>
               'fullname': (str) <user.profile.name>
            }
        }
        """
       User = get_user_model()

    users = _use_read_replica(
        User.objects.filter(anonymoususerid__anonymous_user_id__in=anonymized_ids)
        .select_related("profile")
        .annotate(
            anonymous_id=F("anonymoususerid__anonymous_user_id")),
            fullname=F("profile__name"),
        )
        .values("username", "email", "fullname", "anonymous_id")
    )

    anonymous_id_to_user_info_mappnig = {
        user["anonymous_id"]: user for user in users
    }
    return anonymous_id_to_user_info_mapping
```

Reply:
I like that idea. If we make this change, we'd also need to modify: https://github.com/openedx/edx-ora2/pull/2076/files#diff-4708da322d3d3b03cd269f5aa9574e7b1708cd6e665f1c25b2ba2570ca18dfefR98-R113 so we keep the implementation consistent